### DISCUSSION TITLE
Embedding data studio reports

### DISCUSSION THREAD

**Discussion by cnilsen** [Geosyntec/StormPiper#121] [Show and tell]
### I made a simple facility report in datastudio 
@acang-gs try this out for embedding in the BMP performance page. 

To request a specific alt-id, encode a json object and append it to the base url. An example is below: 

```javascript
var params = {
  "tmnt.id": "SWFA-100006"
};

//turn params into a string
var paramsAsString = JSON.stringify(params);

//encode the url
var encodedParams = encodeURIComponent(paramsAsString)
```
This results in the following encoded string:

```javascript
"%7B%22tmnt.id%22%3A%20%22SWFA-100006%22%7D"
```
Append this to the base url for the facility page: 
```javascript
var base_url = "https://datastudio.google.com/embed/reporting/f8c0c67a-4560-47e6-a35a-f4098f1abbc5/page/4ICDD"

var encoded_url = base_url.concat('?params=',encodeParams)
```
The result is the encoded url to request the facility specific report: 
```
"https://datastudio.google.com/embed/reporting/f8c0c67a-4560-47e6-a35a-f4098f1abbc5/page/4ICDD?params=%7B%22tmnt.id%22%3A%20%22SWFA-100006%22%7D"
```

Result page for embedding: 

<img width="300" alt="image" src="https://user-images.githubusercontent.com/4753248/213828727-baeb3398-47db-4933-932b-6202b3ce7250.png">

**Comment by cnilsen** [comment]
@austinorr

**Reply by austinorr** [reply]
Cool I’m glad the datastudio accepts flexible query params. 

I’m confused about the need for this process though, can you share some info about why this is an embed request rather than a style request for our existing tract component? 

My understanding is that making additional 3rd party request handlers is much harder than adding 3-4 facility info fields to our existing component.

**Reply by austinorr** [reply]
@cnilsen We really should try to use data studio as sparingly as possible. It’s not version controlled, and it likely doesn’t know the difference between the dev site and the prod site, which have different databases. Also you’re the only one with access, so if something goes wrong you can’t delegate the edit.

As far as I can tell, this specific example is redundant since the frontend already makes the api call to fetch all the tmnt details and results. React is the frontend renderer we chose, not datastudio, so we should stick with it unless it’s _impossible_ or at least _difficult_ to do with react and our backend (eg the bigquery land use summary).

**Reply by cnilsen** [reply]
That's fine with me. This is just a quick example to try out. I agree with this: 

> we should stick with it unless it’s impossible or at least difficult to do with react and our backend (eg the bigquery land use summary).
